src/DIRAC/FrameworkSystem/Agent/ComponentSupervisionAgent.py

```diff
@@ -0,0 +1,549 @@
+      return resRunning
+    if not resStopped['OK']:
+      return resStopped
+    defaultStatus['Run'] = set(resRunning['Value'])
+    defaultStatus['Down'] = set(resStopped['Value'])
+    defaultStatus['All'] = defaultStatus['Run'] | defaultStatus['Down']
+
+    if defaultStatus['Run'].intersection(defaultStatus['Down']):
+      self.logError('Overlap in configuration', str(defaultStatus['Run'].intersection(defaultStatus['Down'])))
+      return S_ERROR('Bad host configuration')
+
+    return S_OK(defaultStatus)
+
+  def _ensureComponentRunning(self, shouldBeRunning):
+    """Ensure the correct components are running."""
+    for instance in shouldBeRunning:
+      self.log.info('Starting instance %s' % instance)
+      system, name = instance.split('__')
+      if self.controlComponents:
+        res = self.sysAdminClient.startComponent(system, name)
+        if not res['OK']:
+          self.logError('Failed to start component:', '%s: %s' % (instance, res['Message']))
+        else:
+          self.accounting[instance]['Treatment'] = 'Instance was down, started instance'
+      else:
+        self.accounting[instance]['Treatment'] = 'Instance is down, should be started'
+
+  def _ensureComponentDown(self, shouldBeDown):
+    """Ensure the correct components are not running."""
+    for instance in shouldBeDown:
+      self.log.info('Stopping instance %s' % instance)
+      system, name = instance.split('__')
+      if self.controlComponents:
+        res = self.sysAdminClient.stopComponent(system, name)
+        if not res['OK']:
+          self.logError('Failed to stop component:', '%s: %s' % (instance, res['Message']))
+        else:
+          self.accounting[instance]['Treatment'] = 'Instance was running, stopped instance'
+      else:
+        self.accounting[instance]['Treatment'] = 'Instance is running, should be stopped'
+
+  def checkURLs(self):
+    """Ensure that the running services have their URL in the Config."""
+    self.log.info('Checking URLs')
+    # get services again, in case they were started/stop in controlComponents
+    gConfig.forceRefresh(fromMaster=True)
+    res = self.getRunningInstances(instanceType='Services', runitStatus='All')
+    if not res['OK']:
+      return S_ERROR('Failure to get running services')
+    self.services = res['Value']
+    for service, options in sorted(self.services.items()):
+      self.log.debug('Checking URL for %s with options %s' % (service, options))
+      # ignore SystemAdministrator, does not have URLs
+      if 'SystemAdministrator' in service:
+        continue
+      self._checkServiceURL(service, options)
+
+    if self.csAPI.csModified and self.commitURLs:
+      self.log.info('Commiting changes to the CS')
+      result = self.csAPI.commit()
+      if not result['OK']:
+        self.logError('Commit to CS failed', result['Message'])
+        return S_ERROR('Failed to commit to CS')
+    return S_OK()
+
+  def _checkServiceURL(self, serviceName, options):
```

Comment:
Can you have an option to disable this ? I like the possibility to run services on box without having them used unless I need them to debug them

Reply:
There is the option "commitURLs" that disables changing the URLs in the CS, but you mean an option to completely disable checking the URLs?